contrib/cryptomb/private_key_providers/test/ops_test.cc

```diff
@@ -430,6 +458,105 @@ TEST(CryptoMbProviderTest, TestRSATimer) {
   EXPECT_EQ(res, ssl_private_key_failure);
 }
 
+TEST(CryptoMbProviderTest, TestRSAQueueSizeStatistics) {
+  Event::SimulatedTimeSystem time_system;
+  Stats::TestUtil::TestStore server_stats_store;
+  Api::ApiPtr api = Api::createApiForTest(server_stats_store, time_system);
+  Event::DispatcherPtr dispatcher(api->allocateDispatcher("test_thread"));
+  bssl::UniquePtr<EVP_PKEY> pkey = makeRsaKey();
+  std::shared_ptr<FakeIppCryptoImpl> fakeIpp = std::make_shared<FakeIppCryptoImpl>(true);
+  RSA* rsa = EVP_PKEY_get0_RSA(pkey.get());
+  const BIGNUM *e, *n, *d;
+  RSA_get0_key(rsa, &n, &e, &d);
+  fakeIpp->setRsaKey(n, e, d);
+  CryptoMbStats stats(server_stats_store, CryptoMbQueue::MULTIBUFF_BATCH, kStatsPrefix,
+                      kQueueSizeStatPrefix);
+
+  CryptoMbQueue queue(std::chrono::milliseconds(200), KeyType::Rsa, 1024, fakeIpp, *dispatcher,
+                      stats);
+
+  size_t in_len = 32;
+  uint8_t in[32] = {0x7f};
```

Comment:
there's a lot of repeated blocks of code in this test file. WDYT of factoring some of those out into methods/members of a test class for CryptoMbProviderTest, and shortening all these tests to hopefully make them more readable?

Then we can document some of the code that's mysterious like what this 0x7f is supposed to represent, or why `in` is 32 bytes long.

Reply:
Refactored tests in ops_test.cc and added comments.